src/Facets/EcoFacet.sol

```diff
@@ -0,0 +1,256 @@
+        uint256 nativeFeeAmount = LibAsset.isNativeAsset(
+            _bridgeData.sendingAssetId
+        )
+            ? _ecoData.solverReward
+            : 0;
+
+        _bridgeData.minAmount = _depositAndSwap(
+            _bridgeData.transactionId,
+            _bridgeData.minAmount,
+            _swapData,
+            payable(msg.sender),
+            nativeFeeAmount
+        );
+
+        // For ERC20 tokens, the swap result includes the solver reward
+        // We need to subtract it to get the actual bridge amount
+        if (!LibAsset.isNativeAsset(_bridgeData.sendingAssetId)) {
+            _bridgeData.minAmount =
+                _bridgeData.minAmount -
+                _ecoData.solverReward;
+        }
+
+        _startBridge(_bridgeData, _ecoData);
+    }
+
+    /// Internal Methods ///
+
+
+    function _getEcoChainId(
+        uint256 _lifiChainId
+    ) private pure returns (uint64) {
+        // Map LiFi Tron chain ID to Eco protocol Tron chain ID
+        if (_lifiChainId == LIFI_CHAIN_ID_TRON) {
+            return 728126428; // Eco protocol's Tron chain ID
+        }
+        // Map LiFi Solana chain ID to Eco protocol Solana chain ID
+        if (_lifiChainId == LIFI_CHAIN_ID_SOLANA) {
+            return 1399811149; // Eco protocol's Solana chain ID
+        }
+        // For all other chains (EVM), pass through as-is
+        return uint64(_lifiChainId);
+    }
+
+    function _buildReward(
+        ILiFi.BridgeData memory _bridgeData,
+        EcoData calldata _ecoData
+    ) private view returns (IEcoPortal.Reward memory) {
+        bool isNative = LibAsset.isNativeAsset(_bridgeData.sendingAssetId);
+
+        IEcoPortal.TokenAmount[] memory rewardTokens;
+        if (!isNative) {
+            rewardTokens = new IEcoPortal.TokenAmount[](1);
+            rewardTokens[0] = IEcoPortal.TokenAmount({
+                token: _bridgeData.sendingAssetId,
+                amount: _bridgeData.minAmount + _ecoData.solverReward
+            });
+        } else {
+            rewardTokens = new IEcoPortal.TokenAmount[](0);
+        }
+
+        return
+            IEcoPortal.Reward({
+                creator: msg.sender,
+                prover: _ecoData.prover,
+                deadline: _ecoData.rewardDeadline,
+                nativeAmount: isNative
+                    ? _ecoData.solverReward + _bridgeData.minAmount
+                    : 0,
+                tokens: rewardTokens
+            });
+    }
+
+    function _startBridge(
+        ILiFi.BridgeData memory _bridgeData,
+        EcoData calldata _ecoData
+    ) internal {
+        // Build reward for the intent
+        IEcoPortal.Reward memory reward = _buildReward(_bridgeData, _ecoData);
+
+        // Get the destination chain ID in Eco format
+        uint64 destination = _getEcoChainId(_bridgeData.destinationChainId);
+
+        // Determine if sending native tokens and calculate value
+        bool isNative = LibAsset.isNativeAsset(_bridgeData.sendingAssetId);
```

Comment:
wondering if it could make sense to move this line to the beginning of the function and pass it as a parameter to _buildReward so that we dont need to check isNativeAsset several times. But not sure if that would really save any gas.
The same goes for totalValue...this could also be passed to _buildReward as it's calculated there again.

Reply:
fixed: bd753870